backend/lib/endpoints/posts.js

```diff
@@ -215,8 +221,9 @@ async function routes(app) {
     { preValidation: [app.authenticate], schema: likeUnlikePostSchema },
     async (req) => {
       const { postId, userId } = req.params;
-      // todo: get user id from JWT
-      //  check if userId is the same as param, and if authorized to like (based on visibility)
+      if (!req.userId.equals(userId)) {
```

Comment:
I wonder if we could make this DRY by moving this code to a prevalidation hook? https://www.fastify.io/docs/latest/Hooks/

Reply:
yeah I was also thinking about extracting some of the repeating code blocks here, wasn't sure yet what the best way would be, we started with that `backend/components` folder which seemed to be intended for reusable blocks but those functions are more difficult to handle erorrs (if we have to `throw new Error(...)` or `throw new HttpError(...)` 
so I was kind of thinking about a more systematic approach, perhaps using more hooks will be the best way to do it

Reply:
Also, I think it would be better to use `preHandler` so the order would be :
- check auth first (decode JWT + add user on req object)
- schema validation is done to ensure the request has correct format
- any additional hooks could do things like check if a user is only updating his / her own resource. But I think these patterns are not very established at the moment, in a RESTful way we could use PUT with own user Id in the route param or only use the `req.userId` and always return 200 response which is maybe better for idempotent REST request (PUT, PATCH or DELETE) so the actual auth check can differ for each operation